README.md

```diff
@@ -35,6 +35,7 @@ Certain properties of a resource are _semantic_ and have special meaning when us
 * **`writeOnlyProperties`**: A property in the `writeOnlyProperties` cannot be returned in a **READ** or **LIST** request, and can be used to express things like passwords, secrets or other sensitive data.
 * **`createOnlyProperties`**: A property in the `createOnlyProperties` cannot be specified in an **UPDATE** request, and can only be specified in a **CREATE** request. Another way to think about this - these are properties which are 'write-once', such as the [`Engine`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-engine) property for an [`AWS::RDS::DBInstance`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html) and if you wish to change such a property on a live resource, you should replace that resource by creating a new instance of the resource and terminating the old one. This is the behaviour CloudFormation follows for all properties documented as _'Update Requires: Replacement'_. An attempt to supply these properties to an **UPDATE** request will produce a runtime error from the handler.
 * **`deprecatedProperties`**: A property in the `deprecatedProperties` is not guaranteed to be present in the response from a **READ** request. These fields will still be accepted as input to **CREATE** and **UPDATE** requests however they may be ignored, or converted to new API forms when outbound service calls are made.
+* **`replacementStrategy`**: Used for resource that needs replacement upon an update, where CloudFormation recreates the resource and generates a new physical ID. Must be either `[create, delete]` or `[delete, create]`. Default value is `[create, delete]`
```

Comment:
I think the description here is a bit confusing, and we want to give a bit more context.  Let me give it a try...

"""
As mentioned above, changing a `createOnlyProperty` requires replacement of the resource by creating a new one and deleting the old one.  The default CloudFormation replacement behavior is to create a new resource first, then delete the old resource, so as to avoid any downtime. However, some resources are singleton resources, meaning that only one can exist at a time. In this case, it is not possible to create a second resource first, so CloudFormation must Delete first and then Create.  Specify either `[create, delete]` or `[delete, create]`. Default value is `[create, delete]`
"""

Now that I'm writing that, this does seem to be bleeding into CloudFormation specific behavior.  Are we okay with that @rjlohan?  

I think the bigger concern here is that without the context of CloudFormation, this doesn't really mean anything.  Someone might look at their resource and say "I dunno, it doesn't really matter whether you create then delete or delete then create. You could do it either way".  This lends me to think maybe this property should take some other form.

That's why I like the idea of `isSingletonResource` or some other boolean value that is not open to interpretation.

Reply:
I think this is OK, and it's not really a CloudFormation behaviour, it's a function of how a resource can be safely replaced which I think is more general. I also prefer that we use some explicit naming rather than a flag and name that has to be interpreted. `isSingletonResource` doesn't tell me anything about what this field does at runtime. I think the `replacementStrategy` naming is clearer here, and the values are more expicit.

Reply:
I'm OK with the text update though. 👍